File: HumanEval/125.py
## BEGIN - PROMPT

def split_words(txt):
    '''
    Given a string of words, return a list of words split on whitespace, if no whitespaces exists in the text you
    should split on commas ',' if no commas exists you should return the number of lower-case letters with odd order in the
    alphabet, ord('a') = 0, ord('b') = 1, ... ord('z') = 25
    Examples
    split_words("Hello world!") ➞ ["Hello", "world!"]
    split_words("Hello,world!") ➞ ["Hello", "world!"]
    split_words("abcdef") == 3 
    '''

## BEGIN - CANONICAL SOLUTION
    if " " in txt:
        return txt.split()
    elif "," in txt:
        return txt.replace(',',' ').split()
    else:
        return len([i for i in txt if i.islower() and ord(i)%2 == 0])

## BEGIN - TEST
def check(candidate):

    assert candidate("Hello world!") == ["Hello","world!"]
    assert candidate("Hello,world!") == ["Hello","world!"]
    assert candidate("Hello world,!") == ["Hello","world,!"]
    assert candidate("Hello,Hello,world !") == ["Hello,Hello,world","!"]
    assert candidate("abcdef") == 3
    assert candidate("aaabb") == 2
    assert candidate("aaaBb") == 1
    assert candidate("") == 0

## BEGIN - CHECK
check(split_words)

File: HumanEvalJs/125.js
//// BEGIN - PROMPT
/**
 * Given a string of words, return an array of words split on whitespace.
 * If no whitespace exists, split on commas ','.
 * If no commas exist, return the number of lowercase letters whose ASCII code is even.
 * ASCII codes: 'a' = 97, 'b' = 98, ..., 'z' = 122
 *
 * Examples:
 * splitWords("Hello world!") ➞ ["Hello", "world!"]
 * splitWords("Hello,world!") ➞ ["Hello", "world!"]
 * splitWords("abcdef") ➞ 3
 *
 * @param {string} txt
 * @returns {string[]|number}
 */

function splitWords(txt) {
//// BEGIN - CANONICAL SOLUTION
    if (txt.includes(' ')) {
        return txt.split(/\s+/);
    } else if (txt.includes(',')) {
        return txt.replace(/,/g, ' ').split(/\s+/);
    } else {
        let count = 0;
        for (let char of txt) {
            const code = char.charCodeAt(0);
            if (char >= 'a' && char <= 'z' && code % 2 === 0) {
                count++;
            }
        }
        return count;
    }
}

//// BEGIN - TEST
const METADATA = {
    author: 'jt',
    dataset: 'test'
};

const assert = require('assert');
function check(candidate) {
    assert.deepStrictEqual(candidate("Hello world!"), ["Hello", "world!"]);
    assert.deepStrictEqual(candidate("Hello,world!"), ["Hello", "world!"]);
    assert.deepStrictEqual(candidate("Hello world,!"), ["Hello", "world,!"]);
    assert.deepStrictEqual(candidate("Hello,Hello,world !"), ["Hello,Hello,world", "!"]);
    assert.strictEqual(candidate("abcdef"), 3);
    assert.strictEqual(candidate("aaabb"), 2);
    assert.strictEqual(candidate("aaaBb"), 1);
    assert.strictEqual(candidate(""), 0);
}

//// BEGIN - CHECK
check(splitWords);